pkg/util/env_test.go

```diff
@@ -4,60 +4,54 @@ import (
+		{Input: "HTTPS_PROXY=https://user:password@hostname.com:8443", Output: "HTTPS_PROXY=https://redacted@hostname.com:8443"},
+		// values with no user+password info are untouched
+		{Input: "http_proxy=http://hostname.com", Output: "http_proxy=http://hostname.com"},
+		{Input: "https_proxy=https://hostname.com", Output: "https_proxy=https://hostname.com"},
+		{Input: "HTTP_PROXY=http://hostname.com", Output: "HTTP_PROXY=http://hostname.com"},
+		{Input: "HTTPS_PROXY=https://hostname.com", Output: "HTTPS_PROXY=https://hostname.com"},
+		{Input: "http_proxy=http://user@hostname.com", Output: "http_proxy=http://user@hostname.com"},
+		{Input: "https_proxy=https://user@hostname.com", Output: "https_proxy=https://user@hostname.com"},
+		{Input: "HTTP_PROXY=http://user@hostname.com", Output: "HTTP_PROXY=http://user@hostname.com"},
+		{Input: "HTTPS_PROXY=https://user@hostname.com", Output: "HTTPS_PROXY=https://user@hostname.com"},
 		// keys that don't contain "proxy" are untouched
-		"othervalue=http://user:password@hostname.com",
-		"othervalue=user:password@hostname.com",
+		{Input: "othervalue=http://user:password@hostname.com", Output: "othervalue=http://user:password@hostname.com"},
+		{Input: "othervalue=user:password@hostname.com", Output: "othervalue=user:password@hostname.com"},
 		// unparseable url
-		"proxy=https://user:password@foo%$ @bar@blah.com",
+		{Input: "proxy=https://user:password@foo%$ @bar@blah.com", Output: "proxy=https://user:password@foo%$ @bar@blah.com"},
 	}
 
-	expected := []string{
-		"http_proxy=user:password@hostname.com",
-		"https_proxy=user:password@hostname.com",
-		"HTTP_PROXY=user:password@hostname.com",
-		"HTTPS_PROXY=user:password@hostname.com",
-		"http_proxy=http://redacted@hostname.com",
-		"https_proxy=https://redacted@hostname.com",
-		"HTTP_PROXY=http://redacted@hostname.com",
-		"HTTPS_PROXY=https://redacted@hostname.com",
-		"http_proxy=http://redacted@hostname.com:80",
-		"https_proxy=https://redacted@hostname.com:443",
-		"HTTP_PROXY=http://redacted@hostname.com:8080",
-		"HTTPS_PROXY=https://redacted@hostname.com:8443",
-		"http_proxy=http://redacted@hostname.com",
-		"https_proxy=https://redacted@hostname.com",
-		"HTTP_PROXY=http://redacted@hostname.com",
-		"HTTPS_PROXY=https://redacted@hostname.com",
-		"othervalue=http://user:password@hostname.com",
-		"othervalue=user:password@hostname.com",
-		"proxy=https://user:password@foo%$ @bar@blah.com",
+	inputs := make([]string, len(cases))
+	expected := make([]string, len(cases))
+	for i, c := range cases {
+		inputs[i] = c.Input
+		expected[i] = c.Output
 	}
 	result := SafeForLoggingEnv(inputs)
 	for i := range result {
 		if result[i] != expected[i] {
-			t.Errorf("expected %s to be stripped to %s, but got %s", inputs[i], expected[i], result[i])
+			t.Errorf("expected %s to be changed to %s, but got %s", inputs[i], expected[i], result[i])
```

Comment:
`expected %s to be %s`  (this test no longer expects the string to be changed in all cases, there are some cases where it should be untouched)